src/backend/cdb/cdbmutate.c

```diff
@@ -2406,20 +2426,24 @@ shareinput_mutator_xslice_1(Node *node, PlannerInfo *root, bool fPop)
 		Motion	   *motion = (Motion *) plan;
 
 		shareinput_pushmot(ctxt, motion->motionID);
+
+		/* Top node of subplan should have a Flow node. */
+		Insist(motion->plan.lefttree && motion->plan.lefttree->flow);
```

Comment:
One of my PRs to upstream wasn't approved, because it contained Insist macro. The guys from upstream want to avoid using it for a reason, so, we should avoid its usage too.

Reply:
> One of my PRs to upstream wasn't approved, because it contained Insist macro. The guys from upstream want to avoid using it for a reason, so, we should avoid its usage too.

I copy it from here https://github.com/arenadata/gpdb/blob/e69509affc7d96b7298b9ae80582c974cd4b0d4a/src/backend/executor/execMain.c#L5126-L5127

Reply:
So what? There are multiple examples of using Insist in code.

Reply:
may be use `AssertImply`?

Reply:
I don't think it fit your needs. But maybe you can show the way to implement the same condition using AssertImply correctly.

Reply:
> you can show the way to implement the same condition using AssertImply correctly

```c
AssertImply(motion->plan.lefttree, motion->plan.lefttree->flow); 
```
?

Reply:
Will assertion fire if motion->plan.lefttree is NULL?

Reply:
or simply
```c
Assert(motion->plan.lefttree);
Assert(motion->plan.lefttree->flow); 
```

Reply:
Yes, that's one of options.

Reply:
done